dispatch/api/serializers.py

```diff
@@ -455,8 +456,90 @@ def load_data(self):
 
     def insert_data(self, content):
         """Insert loaded data into embed data blocks."""
+
         return map(self.insert_instance, content)
 
+class ColumnArticleSerializer(DispatchModelSerializer):
+    """Serializes articles for the Column model"""
+    id = serializers.ReadOnlyField(source='parent_id')
+    authors = AuthorSerializer(many=True, read_only=True)
+
+    class Meta:
+        model = Article
+        fields = (
+            'id',
+            'headline',
+            'authors',
+            'published_version'
+        )
+
+class ColumnSerializer(DispatchModelSerializer):
+    """Serializes the Column model"""
+
+    authors = AuthorSerializer(many=True, read_only=True)
+    author_ids = serializers.ListField(
+        write_only=True,
+        child=serializers.JSONField(),
+        validators=[AuthorValidator]
+    )
+    authors_string = serializers.CharField(source='get_author_string', read_only=True)
+    featured_image = ImageAttachmentSerializer(required=False, allow_null=True)
+    articles = ColumnArticleSerializer(many=True, read_only=True, source='get_articles')
+    article_ids = serializers.ListField(
+        write_only=True,
+        child=serializers.JSONField(),
+        required=False
+    )
+    section = SectionSerializer(read_only=True)
+    section_id = serializers.IntegerField(write_only=True)
+
+    slug = serializers.SlugField(validators=[ColumnSlugValidator()])
+
+    class Meta:
+        model = Column
+        fields = (
+            'id',
+            'name',
+            'section',
+            'section_id',
+            'slug',
+            'description',
+            'featured_image',
+            'authors',
+            'author_ids',
+            'authors_string',
+            'articles',
+            'article_ids'
+        )
+
+    def create(self, validated_data):
+        instance = Column()
+        return self.update(instance, validated_data)
+
+    def update(self, instance, validated_data):
+        # Update basic fields
+        instance.name = validated_data.get('name', instance.name)
+        instance.slug = validated_data.get('slug', instance.slug)
+        instance.section_id = validated_data.get('section_id', instance.section_id)
+        instance.description = validated_data.get('description', instance.description)
+
+        # Save instance before processing/saving content in order to save associations to correct ID
+        instance.save()
+
+        # featured_image = validated_data.get('featured_image', False)
+        # if featured_image != False:
+        #     instance.save_featured_image(featured_image)
```

Comment:
Why is this commented out?

Reply:
I wasn't quite sure at the time whether we would end up using featured images for columns and forgot to resolve it. What do you think? For the front end I ended up copying the section template fairly closely